payments/providers/bambora_payform.py

```diff
@@ -254,6 +264,102 @@ def payload_add_auth_code(self, payload: dict):
         data = "{}|{}".format(payload["api_key"], payload["order_number"])
         payload.update(authcode=self.calculate_auth_code(data))
 
+    def initiate_refund(self, order: Order) -> OrderRefund:
+        # Orders which are PAID_MANUALLY may not have an entry in Bambora,
+        # so we can't guarantee that the refund will be executed
+        if order.status != OrderStatus.PAID:
+            raise ValidationError(
+                "Cannot refund an order that has not been paid through VismaPay"
+            )
+
+        order_token = (
+            order.tokens.exclude(token__isnull=True, token__iexact="")
+            .order_by("-created_at")
+            .first()
+        )
+
+        if not order_token:
+            raise ValidationError(
+                "Cannot refund an order that has not been paid through VismaPay"
+            )
+
+        payload = {
+            "version": "w3.1",
+            "api_key": self.config.get(VENE_PAYMENTS_BAMBORA_API_KEY),
+            "amount": price_as_fractional_int(order.price),
+            "notify_url": self.get_notify_refund_url(),
+            "order_number": f"{order.order_number}-{order_token.created_at.timestamp()}",
+        }
+        self.payload_add_auth_code(payload)
+
+        try:
+            r = requests.post(self.url_refund, json=payload, timeout=60)
```

Comment:
> _Visma docs on /pbwapi/create_refund say: "Refund payment. Please contact our support if you want to use this feature." -> has their support been contacted?_

Yes, I contacted them yesterday :)

> _Should the timeout be shorter? Our front-end nginx will time out after 60 seconds. It would be good to have something like 50 seconds here, so in case it times out, the admin will at least get a better error message?_

Hmm I guess there's no harm in changing it, it _shouldn't_ take that long anyway.